source/wp-content/themes/wporg-developer-2023/src/command-github/block.php

```diff
@@ -0,0 +1,143 @@
+<?php
+namespace WordPressdotorg\Theme\Developer_2023\Dynamic_Command_Github;
+
+add_action( 'init', __NAMESPACE__ . '\init' );
+
+/**
+ * Registers the block using the metadata loaded from the `block.json` file.
+ * Behind the scenes, it registers also all assets so they can be enqueued
+ * through the block editor in the corresponding context.
+ *
+ * @see https://developer.wordpress.org/reference/functions/register_block_type/
+ */
+function init() {
+	register_block_type(
+		dirname( dirname( __DIR__ ) ) . '/build/command-github',
+		array(
+			'render_callback' => __NAMESPACE__ . '\render',
+		)
+	);
+}
+
+/**
+ * Render the block content.
+ *
+ * @return string Returns the block markup.
+ */
+function render( $attributes, $content, $block ) {
+
+	$post_ID = $block->context['postId'];
+
+	if ( ! isset( $post_ID ) ) {
+		return '';
+	}
+
+	$repo_url = get_post_meta( $post_ID, 'repo_url', true );
+	$cmd_slug = str_replace( 'wp ', '', get_the_title( $post_ID ) );
+	$open_issues = 'https://github.com/login?return_to=%2Fissues%3Fq%3Dlabel%3A' . urlencode( 'command:' . str_replace( ' ', '-', $cmd_slug ) ) . '+sort%3Aupdated-desc+org%3Awp-cli+is%3Aopen';
+	$closed_issues = 'https://github.com/login?return_to=%2Fissues%3Fq%3Dlabel%3A' . urlencode( 'command:' . str_replace( ' ', '-', $cmd_slug ) ) . '+sort%3Aupdated-desc+org%3Awp-cli+is%3Aclosed';
+	$issue_count = get_issue_count( $cmd_slug );
+
+	if ( $repo_url ) {
+		$content .= sprintf(
+			'
+			<div class="github-tracker">
+				<a href="%s">
+					<img src="https://make.wordpress.org/cli/wp-content/plugins/wporg-cli/assets/images/github-mark.svg" class="icon-github" />
+				</a>
+			',
+			esc_url( $repo_url )
+		);
+	}
+
+	$content .= sprintf(
```

Comment:
I find this section hard to read because some of the tags are closed in the `sprintf` and some aren't. It would be pretty easy to make a mistake and lose the closing `</a` line. 

Could we maybe separate all the `<a>` elements into their own `sprintf` and then just add the containers to the template that is returned:

```
$issue_count = "";
if ( false !== $issue_count['open'] ) {
		$issue_count = sprintf(
			' <span class="green">(%s)</span>',
			(int) $issue_count['open']
		);
}

$content .= sprintf(
		' <a href="%1$s"class="button">%2$s %3$s</a>',
		esc_url( $closed_issues ),
		__( 'View Closed Issues', 'wporg' ),
                 $open_issue,
);
```
...
```
return sprintf(
		'<section %1$s><div class="btn-group">%2$s</div></section>',
		$wrapper_attributes,
		$content,
	);

```

Reply:
I see there are other conditionals, so maybe the example maybe not work exactly.

Reply:
I separated them and tidied it up a bit in [Adjust code format for a better readability](https://github.com/WordPress/wporg-developer/pull/211/commits/b154744be82ac0c577c5f04447ec5ec602325293)